src/IO/WriteBufferFromFileDecorator.cpp

```diff
@@ -31,11 +38,18 @@ WriteBufferFromFileDecorator::~WriteBufferFromFileDecorator()
     {
         tryLogCurrentException(__PRETTY_FUNCTION__);
     }
+
+    swap(*impl);
```

Comment:
What's the point of calling `swap` here? It's after `finalize()`, the buffer isn't going to be used anymore.

Reply:
After finalize call the impl's destructor is called. It is crucial to call it with the right buffers. Not with nullptr's. WriteBufferFromFileDecorator calls swap at constructor. It is has to be call at destructor.

Reply:
Please add a comment saying that.

Reply:
Done